### ISSUE TITLE
Bug? Calendar highlights one day prior to selection date

### ISSUE THREAD

**Issue by apg-dev** [zadam/trilium#845]
I can't figure out why the calendar widget highlights "yesterday's" date, not today. I see this regardless of which date I click on, the day prior to the one selected gets the “calendar-date-active” class setting. Looking at calendar.js, activeDate seems to be off by -1.

<img width="1006" alt="trilium-calendar-highlight" src="https://user-images.githubusercontent.com/55888411/73471776-e87ddd80-4357-11ea-9aa7-116f12abe014.png">
<img width="959" alt="active-date-calendar-js" src="https://user-images.githubusercontent.com/55888411/73517027-4d711c00-43c8-11ea-9f01-fdcd651d5b00.png">

**Comment by zadam** [comment]
Hello, looks like a bug. What timezone are you in? Wondering if it might be caused by that (for me it works fine but I'm close to GMT ...

**Comment by apg-dev** [comment]
EST USA (UTC/GMT - 5 hours). Seems to be independent of the time of day.

**Comment by apg-dev** [comment]
Looks like you're right - it is timezone related. 

I wrote the following code to determine the user's local timezone then convert activeDate to GMT. I tested this in my local environment and it works fine.

In /src/public/javascripts/widgets/calendar.js 

REPLACE line 44: 
'this.activeDate = new Date(Date.parse(activeDate));`

With these four lines:

`var activeDateLocal = new Date(activeDate);
var tzo = activeDateLocal.getTimezoneOffset();
var activeDateGMT = new Date(activeDateLocal.setMinutes( activeDateLocal.getMinutes() + tzo ));
this.activeDate = activeDateGMT;`

Let me know if you'd like me to create a formal patch.
